dynamiqs/qarrays/sparse_dia_qarray.py

```diff
@@ -298,16 +298,16 @@ def _construct_diags(offsets: tuple[int, ...], other: ArrayLike) -> Array:
     return diags
 
 
-def to_sparse(x: DenseQArray | Array) -> SparseQArray:
-    r"""Returns the input matrix in the `SparseQArray` format.
+def to_sparse_dia(x: DenseQArray | Array) -> SparseDIAQArray:
```

Comment:
I am conflicted on this API. WDYT about `def to_sparse(..., format=dq.SparseQArray) -> QArray` or something like that?

Reply:
Meh, why don't you like this version? The other sparse formats may need extra constructor arguments that we are not yet aware of.